BorrowToOwn.API/Controllers/ProductController.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using AutoMapper;
+using BorrowToOwn.Services.Communications.RequestObject.DTO;
+using BorrowToOwn.Services.Communications.ResponseObject.DTO;
+using BorrowToOwn.Services.Contracts;
+using BorrowToOwn.Services.Helpers;
+using Microsoft.AspNetCore.Mvc;
+using static BorrowToOwn.Data.Common.AppEnum;
+
+namespace BorrowToOwn.API.Controllers
+{
+    [ApiController]
+    [Route("/api/Products")]
+    public class ProductController:Controller
+    {
+        private readonly IProductService _productService;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IMapper _mapper;
+
+        public ProductController(IProductService productService,IHttpClientFactory httpClientFactory,IMapper mapper)
+        {
+            _productService = productService ?? throw new ArgumentNullException(nameof(productService));
+            _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+        [HttpPost("")  ,
+         DisableRequestSizeLimit,
+         RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = long.MaxValue)]
+        [Consumes("multipart/form-data")]
```

Comment:
i'm curious, why form data?

Reply:
Because i want files to be uploaded from forms